Source/SumIQ.cpp

```diff
@@ -138,3 +138,137 @@ PeleC::sum_integrated_quantities()
 #endif
   }
 }
+
+void
+PeleC::monitor_extrema()
+{
+  BL_PROFILE("PeleC::sum_integrated_quantities()");
+
+  if (verbose <= 0) {
+    return;
+  }
+
+  bool local_flag = true;
+
+  int finest_level = parent->finestLevel();
+  amrex::Real time = state[State_Type].curTime();
+  amrex::Vector<std::string> extrema_vars = {
+    "density", "x_velocity", "y_velocity", "z_velocity",
+    "eint_e",  "Temp",       "pressure",   "massfrac"};
+
+  int nspec_extrema = 1;
+  if (!fuel_name.empty()) {
+    nspec_extrema += 1;
+    extrema_vars.push_back(fuel_name);
+  }
+  if (!flame_trac_name.empty()) {
+    nspec_extrema += 1;
+    extrema_vars.push_back(flame_trac_name);
+  }
+  if (!extrema_spec_name.empty()) {
+    nspec_extrema += 1;
+    extrema_vars.push_back(extrema_spec_name);
+  }
+  auto nextrema = extrema_vars.size();
+  constexpr amrex::Real neg_huge = std::numeric_limits<amrex::Real>::lowest();
+  constexpr amrex::Real huge = std::numeric_limits<amrex::Real>::max();
+  amrex::Vector<amrex::Real> minima, maxima;
```

Comment:
the next four or five lines can just be put in the construction: `minima(nextrema, huge)`, basically the same thing for `maxima` too.